### PR TITLE
Network layer

### PR DESCRIPTION
**Pull request by harishsaini**

### CODE REVIEW

File: Avingo/Code/UI/Sample/ClockInViewController.swift [hunk 1]
```diff
@@ -7,9 +7,17 @@
 //
 
 import UIKit
+import ReactiveCocoa
+import ReactiveSwift
+import Result
+
+final class ClockInViewController: UIViewController {
+	let viewModel = ClockinViewModel()
```
Comment (ahujamanish) [inline Avingo/Code/UI/Sample/ClockInViewController.swift]: you want to make this private?

Comment (harishsaini) [reply Avingo/Code/UI/Sample/ClockInViewController.swift]: will do!

File: Avingo/Code/Model/CoreData/Extensions/User+CoreData.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	static func create() -> User {
+		let context = CoreDataManager.shared.managedObjectContext
+		let user = User(entity: User.entity(), insertInto: context)
+		CoreDataManager.shared.saveContext()
+		return user
+	}
+
+	static func allUsers(predicate: NSPredicate) -> [User] {
+		let context = CoreDataManager.shared.managedObjectContext
+
```
Comment (tinman-fueled) [inline Avingo/Code/Model/CoreData/Extensions/User+CoreData.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Fueled/avingo-ios/blob/c34a5e955c07b6a0e1d7fd645d5c79e2b264201d/Avingo/Code/Model/CoreData/Extensions/User+CoreData.swift#L22"></span>Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)</td>
    </tr>
  </tbody>
</table>

**Comment by tinman-fueled** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">e73933925c66e6b7e4162b832aecf5c3e9318ccb <code>feat(device-setup): code refactor</code><br />
• This commit’s message is the same as c34a5e955c07b6a0e1d7fd645d5c79e2b264201d. Please consider squashing the commits together.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">c34a5e955c07b6a0e1d7fd645d5c79e2b264201d <code>feat(device-setup): code refactor</code><br />
• This commit’s message is the same as e73933925c66e6b7e4162b832aecf5c3e9318ccb. Please consider squashing the commits together.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**